File: msgraph-mail/typescript/src/users/messagesResponse.ts
import {SerializationWriter, ParseNode, Parsable} from '@microsoft/kiota-abstractions';
import {Message} from './message';

export class MessagesResponse implements Parsable<MessagesResponse> {
    public value?: Message[] | undefined;
    public nextLink?: string | undefined;
    public serialize (writer: SerializationWriter) : void {
        writer.writeCollectionOfObjectValues<Message>("value", this.value);
        writer.writeStringValue("nextLink", this.nextLink);
    };
    public deserializeFields () : Map<string, (item: MessagesResponse, node: ParseNode) => void> {
        return new Map<string, (item: MessagesResponse, node: ParseNode) => void>([
            ["value", (o, n) => { o.value = n.getCollectionOfObjectValues<Message>(Message); }],
            ["nextLink", (o, n) => { o.nextLink = n.getStringValue(); }],
        ]);
    };
}

File: msgraph-mail/dotnet/Users/MessagesResponse.cs
using Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.Linq;
namespace Graphdotnetv4.Users {
    public class MessagesResponse : IParsable<MessagesResponse> {
        public List<Message> Value { get; set; }
        public string NextLink { get; set; }
        public IDictionary<string, Action<MessagesResponse, IParseNode>> DeserializeFields => new Dictionary<string, Action<MessagesResponse, IParseNode>> {
            {
                "value", (o,n) => { o.Value = n.GetCollectionOfObjectValues<Message>().ToList(); }
            },
            {
                "nextLink", (o,n) => { o.NextLink = n.GetStringValue(); }
            },
        };
        public void Serialize(ISerializationWriter writer) {
            writer.WriteCollectionOfObjectValues<Message>("value", Value);
            writer.WriteStringValue("nextLink", NextLink);
        }
    }
}
